incubator/hnc/doc/metrics/stackdriver.md

```diff
@@ -0,0 +1,85 @@
+     ```
+     gcloud iam service-accounts add-iam-policy-binding \
+       --role roles/iam.workloadIdentityUser \
+       --member "serviceAccount:[PROJECT_ID].svc.id.goog[hnc-system/default]" \
+       [GSA_NAME]@[PROJECT_ID].iam.gserviceaccount.com
+   ```
+4. Add the `iam.gke.io/gcp-service-account=[GSA_NAME]@[PROJECT_ID]` annotation to the KSA, using the email address of the Google service account:
+     ```
+     kubectl annotate serviceaccount \
+       --namespace hnc-system \
+       default \
+       iam.gke.io/gcp-service-account=[GSA_NAME]@[PROJECT_ID].iam.gserviceaccount.com
+   ```
+5. Verify the service accounts are configured correctly by creating a Pod with the Kubernetes service account that runs the `cloud-sdk` container image, and connecting to it with an interactive session:
+     ```
+     kubectl run --rm -it \
+       --generator=run-pod/v1 \
+       --image google/cloud-sdk:slim \
+       --serviceaccount default \
+       --namespace hnc-system \
+       workload-identity-test
+   ```
+6. You are now connected to an interactive shell within the created Pod. Run the following command:
+     ```
+     gcloud auth list
+   ```
+   If the service accounts are correctly configured, the GSA email address is listed as the active (and only) identity. 
+
+### Option 2: Use GCE default Service Account
+By default, GKE clusters without Workload Identity use the GCE default Service Account, and since this SA already has permission to write metrics to Stackdriver, no extra steps are required other than creating a GKE cluster and installing HNC. The HNC workloads running on the GCE nodes will use their Service Accounts by default.
+
+## View HNC Metrics in Stackdriver
+You should now be able to view all HNC metrics in Stackdriver [Metrics Explorer](https://cloud.google.com/monitoring/charts/metrics-explorer) by searching the metrics keywords:
+
+|Metric                                              |Description   |
+|:-------------------------------------------------- |:-------------|
+| hnc/reconcilers/hierconfig/total                   | The total number of HierConfig reconciliations happened | 
+| hnc/reconcilers/hierconfig/concurrent_peak         | The peak concurrent HierConfig reconciliations happened in the past 60s, which is also the minimum Stackdriver reporting period and the one we're using |
+| hnc/reconcilers/hierconfig/hierconfig_writes_total | The number of HierConfig writes happened during HierConfig reconciliations | 
+| hnc/reconcilers/hierconfig/namespace_writes_total  | The number of namespace writes happened during HierConfig reconciliations | 
+| hnc/reconcilers/object/total                       | The total number of object reconciliations happened |
+| hnc/reconcilers/object/concurrent_peak             | The peak concurrent object reconciliations happened in the past 60s, which is also the minimum Stackdriver reporting period and the one we're using |
```

Comment:
Maybe move this section a "main" page (eg README in this directory) followed by:

* Instructions for Stackdriver on GKE (link to the rest of this doc)
* Instructions for Prometheus (see #433)

wdyt?

This would also make it easier to expand on the documentation for individual metrics if necessary.

Reply:
Also, if you do this, you can simplify the structure of this doc - e.g. get rid of the H2 "How to set up" heading and promote all other headings by one level (eg "Option 1: Use Workload Identity" gets upgraded from H3 to H2).

Reply:
Sg. I moved the `Viewing metrics` section into `Using HNC` section in the README.